### ISSUE TITLE
Simple brightness adaptions (bias) to survey images

### ISSUE THREAD

**Issue by gzotti** [Stellarium/stellarium#3314]
I only forward this:

### Discussed in https://github.com/Stellarium/stellarium/discussions/3313

<div type='discussions-op-text'>

<sup>Originally posted by **GerritErdt** July  6, 2023</sup>
Hi all, 

today I noticed that the DSS2 Red survey, which I use most of the time to identify faint nebulas, shows much more faint detail when the atmosphere is active and the sky is not yet completely dark. That way, I can clearly see nebula structures that won't show without the additional brightness due to the atmosphere. 

To put it in technical terms: adding a bias to the survey data greatly improves the visibility of faint details. 

Is it possible to include such a custom adjustable bias into future versions for all the surveys? Or maybe even a simple, custom histogram stretch? Event the first, much easier option, would help to better see faint nebulas and thus enhance the planning capabilities stellarium offers. 

CS Gerrit</div>

Some tweaks could be made. Also combination of three B/W surveys to a colored one, or doing some per-survey enhancement.

**Comment by GerritErdt** [comment]
Regarding pre-survey enhancement, I don't know how useful that will be. 

Everyone has different uses for the survey data. By applying a single processing to the data, it will become harder to achieve some unusual goals with the modified data. 

What about offering the option of an offline version of the mentioned survey? That way, it would be possible for everyone to apply their own automated processing to the data. Of course that needs some higher quality data compared to JPGs, but that should be manageable. 

But one thing that would surely help everyone regarding general preprocessing is some form of gradient removal of the survey data. These gradients really suck. I once tried that with the offline set of the DSS survey, but it completely failed due to the data being only JPGs. 

CS Gerrit

**Comment by gzotti** [comment]
per-survey, not pre-survey! E.g., add columns to the list with bias (offset) and enhancement factor (just what you want, bur _per_ each survey individually), and remember them. However, none of those are likely to happen if nobody else steps in. There are always more ideas than dev time.

You can host your own HiPS surveys. See User Guide, 10.2.

**Comment by GerritErdt** [comment]
Oh well, seems like I miss-read your post....

A bias that is remembered for every survey would surely be convenient. But for simplicity reasons, even just a small slider on the screen that does not remember values would be helpful. I guess in practice, the bias needs to be different for different parts of the sky, so I am not really sure if saving the bias is actually beneficial to just manually setting it each time. And it would make development much easier. Just adding a bias to the images seems like a pretty easy operation, at least without knowing the exact internal handling of the survey data. 

Regarding the hosting of my own version of a HiPS survey, it seems possible to use any data in the same way as the offline TOAST survey, as described in part 10.4.1 in the manual. Is that correct? If so, is it necessary to have tiles with the same dimensions as the original TOAST survey data? In general, this seems easier than hosting my own survey as in part 10.2, at least to me. 

In case this is all possible, I "only" have to find a way to generate my own processed data of the DSS2 Red survey and prepare them for usage in the same way as the TOAST survey. Seems like a possible solution to my issues. 

CS Gerrit

**Comment by alex-w** [comment]
> HiPS and TOAST are different file structures. If you make a TOAST yourself, you can host it locally, sure. We don't have special information on the formats, what is not in the User Guide you will have to find online or in your favourite search engine.

I think I've seen tool for creating a TOAST surveys on GitHub :thinking:

**Comment by gzotti** [comment]
Nice! Where is it? We host Guillaume's HiPSter.

**Comment by alex-w** [comment]
> Nice! Where is it? We host Guillaume's HiPSter.

No, it was tool for HiPS - https://github.com/hipspy/hips + https://github.com/hipspy/hips-extra

**Comment by alex-w** [comment]
> Nice! Where is it? We host Guillaume's HiPSter.

I found it! See https://github.com/WorldWideTelescope/toasty

**Comment by GerritErdt** [comment]
Alright guys, thanks, I will see what I can do with that!
